pandas/core/frame.py

```diff
@@ -6667,6 +6667,30 @@ def diff(self, periods: int = 1, axis: Axis = 0) -> "DataFrame":
         5  NaN  NaN   NaN
         """
         bm_axis = self._get_block_manager_axis(axis)
+        self._consolidate_inplace()
```

Comment:
this is not pretty. why are we not simply transposing and calling .diff()?

Reply:
> why are we not simply transposing and calling .diff()?

I'd be fine with that, but it is potentially costly

Reply:
I personally think the column-wise approach could be fine

Reply:
this should be co-located with BlockManager.diff. but to be honest I think transposing is just fine here. The block type is *already* inferred and handled. This is just adding a lot of complexity.

Reply:
if you really want to do this column based approach them move this with the other .diff methods (I actually prefer just a transpose here, its totally fine for now)